tests/auto/main_validator.test.js

```
@@ -0,0 +1,592 @@
+
+// Mock the require calls
+const originalRequire = require;
+const moduleStubs = {
+  './security-validator': function() { return mockSecurityValidator; },
+  './structure-validator': function() { return mockStructureValidator; },
+  './quality-scorer': function() { return mockQualityScorer; },
+  './file-utils': function() { return mockFileUtils; }
+};
+
+// Override require for this test
+require = function(moduleName) {
```

Biome (1.9.4)

[error] A global variable should not be reassigned.

Assigning to a global variable can override essential functionality.

(lint/suspicious/noGlobalAssign)